cmd/srcd/cmd/parse_test.go

```diff
@@ -0,0 +1,257 @@
+		lang:     "python",
+	},
+	{
+		url:      "https://raw.githubusercontent.com/leachim6/hello-world/48eb65bf27272eb7f0887be029b10e7b036b8df3/c/c%2B%2B.cpp",
+		filename: "hello.cpp",
+		lang:     "c++",
+	},
+	{
+		url:      "https://raw.githubusercontent.com/leachim6/hello-world/48eb65bf27272eb7f0887be029b10e7b036b8df3/j/Java.java",
+		filename: "hello.java",
+		lang:     "java",
+	},
+	{
+		url:      "https://raw.githubusercontent.com/leachim6/hello-world/48eb65bf27272eb7f0887be029b10e7b036b8df3/j/javascript.js",
+		filename: "hello.js",
+		lang:     "javascript",
+	},
+	{
+		url:      "https://raw.githubusercontent.com/leachim6/hello-world/48eb65bf27272eb7f0887be029b10e7b036b8df3/b/bash.bash",
+		filename: "hello.bash",
+		lang:     "shell",
+	},
+	{
+		url:      "https://raw.githubusercontent.com/leachim6/hello-world/48eb65bf27272eb7f0887be029b10e7b036b8df3/r/ruby.rb",
+		filename: "hello.rb",
+		lang:     "ruby",
+	},
+	{
+		url:      "https://raw.githubusercontent.com/leachim6/hello-world/48eb65bf27272eb7f0887be029b10e7b036b8df3/g/go.go",
+		filename: "hello.go",
+		lang:     "go",
+	},
+	{
+		url:      "https://raw.githubusercontent.com/leachim6/hello-world/48eb65bf27272eb7f0887be029b10e7b036b8df3/c/csharp.cs",
+		filename: "hello.cs",
+		lang:     "c#",
+	},
+	{
+		url:      "https://raw.githubusercontent.com/leachim6/hello-world/48eb65bf27272eb7f0887be029b10e7b036b8df3/p/php.php",
+		filename: "hello.php",
+		lang:     "php",
+	},
+}
+
+// The suite downloads all the test cases files only once, on the suite setup
+func (s *ParseTestSuite) SetupSuite() {
+	var err error
+	s.testDir, err = ioutil.TempDir("", "parse-test")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Download a hello world file for each supported language
+	for _, tc := range testCases {
+		path := filepath.Join(s.testDir, tc.filename)
+		s.downloadFile(tc.url, path)
+	}
+}
+
+func (s *ParseTestSuite) TearDownSuite() {
+	os.RemoveAll(s.testDir)
+}
+
+func (s *ParseTestSuite) SetupTest() {
+
+}
+
+func (s *ParseTestSuite) TearDownTest() {
+	s.RunStop(context.Background())
+}
+
+func (s *ParseTestSuite) downloadFile(url, path string) {
+	s.T().Helper()
+
+	out, err := os.Create(path)
+	s.Require().NoError(err)
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	s.Require().NoError(err)
+	defer resp.Body.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	s.Require().NoError(err)
+}
+
+func (s *ParseTestSuite) TestDriversList() {
+	require := s.Require()
+
+	out, err := s.RunCommand(context.TODO(), "parse", "drivers", "list")
+	outStr := out.String()
+
+	require.NoError(err, outStr)
+
+	/* Example output:
+
+	LANGUAGE	VERSION
+	----------	----------
+	python		v2.8.0
+	cpp		v1.1.0
+	java		v2.5.0
+	javascript	v2.6.0
+	bash		v2.4.0
+	ruby		v2.9.0
+	go		v2.5.0
+	csharp		v1.4.0
+	php		v2.7.0
+	*/
+
+	// Simple checks to see if it's the table, and contains a known driver
+	require.Contains(outStr, "LANGUAGE	VERSION")
+	expected := regexp.MustCompile(`javascript	v\S+`)
```

Comment:
better `javascript\s+v\S+` because I'm not sure that current regexp will still be valid if there is any new driver with a longer name.

Reply:
Applied and squashed